src/script/script_gui.cpp

```diff
@@ -1254,8 +1261,16 @@ static WindowDesc _script_debug_desc(__FILE__, __LINE__,
 Window *ShowScriptDebugWindow(CompanyID show_company)
 {
 	if (!_networking || _network_server) {
-		ScriptDebugWindow *w = (ScriptDebugWindow *)BringWindowToFrontById(WC_SCRIPT_DEBUG, 0);
-		if (w == nullptr) w = new ScriptDebugWindow(&_script_debug_desc, 0);
+		int i = 0;
+		ScriptDebugWindow *w = nullptr;
+		if (_ctrl_pressed) {
+			/* find next free window number for script debug */
+			while (FindWindowById(WC_SCRIPT_DEBUG, i) != nullptr) i++;
+		} else {
+			w = static_cast<ScriptDebugWindow *>(FindWindowByClass(WC_SCRIPT_DEBUG));
+			if (w != nullptr) BringWindowToFrontById(w->window_class, w->window_number);
+		}
+		if (w == nullptr) w = new ScriptDebugWindow(&_script_debug_desc, i);
```

Comment:
First, if I'm trying to build something and have Ctrl pressed, this would cause each failing script to open a new window. I'm not quite sure that's wanted.
Second, if I have a window for each of the companies, I would expect the failing script to be shown in the window that is already opened for that company. The first script debug window will change company unnecessarily, which would probably be quite annoying if you set up the windows in a specific way. Obviously, when there is no window for that company you can just use the first window it finds.